### DISCUSSION TITLE
Crazyflie behavior on loss of radio link

### DISCUSSION THREAD

**Discussion by enp6s0** [USC-ACTLab/crazyswarm#590] [Q&A] [answered]
If I'm understanding [this](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/crtp/crtp_localization/) correctly, if radio link is lost between the controller and drones, within a few seconds the drones should automatically go into emergency stop because of a missed heartbeat ping, is this correct?

If so, why is this not the (default?) behavior when the drones are operated through Crazyswarm? It appears that when the radio link is lost, the drones just keeps flying (and often drifting away until they crash into something). Could we somehow make use of this fail-safe in a Crazyswarm environment?

Thanks!

**Comment by whoenig** [comment, marked as answer]
You are correct about the heartbeat functionality. Currently, we do not use this feature, but instead send an emergencyStop directly, if requested by the user.

The Crazyswarm supports many different modes of operation, including ones where drones are (semi)-autonomous (e.g., self-localization using AI-deck or LightHouse). For those modes, it may not be desired to keep the regular radio communication up and running.

We are not planning to add new features to Crazyswarm(1) (although, PRs are welcome, of course). However, I open a new issue for Crazyswarm2: https://github.com/IMRCLab/crazyswarm2/issues/18.
